.jenkins/pytorch/build-asan.sh

```diff
@@ -26,7 +26,7 @@ CC="clang" CXX="clang++" LDSHARED="clang --shared" \
   CFLAGS="-fsanitize=address -fsanitize=undefined -fno-sanitize-recover=all -fsanitize-address-use-after-scope -shared-libasan" \
   USE_ASAN=1 USE_CUDA=0 USE_MKLDNN=0 \
   python setup.py bdist_wheel
-  python -mpip install "$(echo dist/*.whl)[opt-einsum]"
+  pip_install_whl "$(echo dist/*.whl)[opt-einsum]"
```

Comment:
Well, this would not work, would it (as the whole point of installing it with `opt-einsum` optional dependency is to pull `opt-einsum` package

Reply:
> `python -mpip install "$(echo dist/*.whl)[opt-einsum]"` downloads and installs `opt-einsum` package, while adding `--nodeps` will breaks that functionality (even though it does not regress CI)

Noted.  Per @janeyx99 comment in https://github.com/pytorch/pytorch/pull/85574#issue-1384342943, let me try to remove `[opt-einsum]` from the Linux workflow:

* We don't need it to be installed here while it's already part of the Docker container
* Also there is no pinned version for the package, which is arguably not the best practice